site/documentation/content/admin-guide/pubsub-config.md

```diff
@@ -0,0 +1,73 @@
++++
+title = "Google Pub/Sub Messaging Configuration"
+weight = 345
++++
+
+Hono can be configured to support Google Pub/Sub as the messaging infrastructure. The Pub/Sub client used for this purpose
+can be configured by means of operating system environment variables.
+
+## Publisher Configuration Properties
+The `org.eclipse.hono.client.pubsub.CachingPubSubPublisherFactory` factory can be used to create Pub/Sub publishers for Hono's
+Pub/Sub based APIs.
+
+The following table provides an overview of the environment variable for configuring a Pub/Sub publisher.
+Note that the variable map to the property of the class `org.eclipse.hono.client.pubsub.PubSubConfigProperties` which is
+used to configure a publisher.
+
+The variable names contain `${PREFIX}` as a placeholder for the particular *common prefix* being used. The `${prefix}`
+placeholder used in the Java system properties is the same as `${PREFIX}`, using all lower case characters and `.`
+instead of `_` as the delimiter, e.g. the environment variable prefix `HONO_PROJECTID` corresponds to the Java system
+property prefix `hono.projectId`.
+
+| OS Environment Variable<br>Java System Property | Mandatory | Default Value | Description  |
+| :---------------------------------------------- | :-------: | :------------ | :------------|
+| `${PREFIX}_PROJECTID`<br>`${prefix}.projectId`   | yes | - | The Project Id of the GCP Project where Pub/Sub is set up. |
```

Comment:
how is the connection to the Google Pub/Sub broker configured? Like host name, port, credentials etc?

Reply:
There is no configurable Google Pub/Sub broker. Pub/Sub is a service which is available in the Google Cloud Platform (GCP). That's why the Google Project Id is always needed.
If someone wants to use Google Pub/Sub, a GCP Project is needed where Pub/Sub is set up/enabled.
To use the Google Pub/Sub Service, the only thing which is needed is a workload identity, but this has to be configured in the GCP Project itself.

Reply:
I do not understand how the Goolge Pub/Sub library will find its cloud endpoint to connect to. Another question is how it finds the credentials to use for authenticating to the Pub/Sub endpoint?
Do how does this info get discovered/conveyed? Does the Google Pub/Sub client library expect this info to be set via an environment variable? A Java system property? A properties file in some well-known location? Does it use the project ID to look up the endpoint dynamically using another Google Cloud API? In any case, pointers to relevant Google Cloud documentation would be helpful here ...